mqtt/handler.go

```diff
@@ -40,3 +45,29 @@ func (h *messageHandler) getMessageHandler() pahomqtt.MessageHandler {
 		h.collector.Observe(h.metric, msg.Topic(), floatValue, labelValues)
 	}
 }
+
+func (h *messageHandler) getJSONMessageHandler() pahomqtt.MessageHandler {
+	return func(_ pahomqtt.Client, msg pahomqtt.Message) {
+		log.Logger.Debugf("Received MQTT msg '%s' from '%s' topic. Listener for: '%s'.", msg.Payload(), msg.Topic(), h.metric.MqttTopic)
+
+		jsonMap := make(map[string]interface{})
+		if err := json.Unmarshal(msg.Payload(), &jsonMap); err != nil {
+			log.Logger.With(zap.Error(err)).Warnf("Got an invalid JSON value '%s' and failed to unmarshal.", msg.Payload())
+			return
+		}
+
+		labelValues := []string{msg.Topic()}
+		for _, idx := range h.metric.TopicLabels {
+			labelValues = append(labelValues, getTopicPart(msg.Topic(), idx))
+		}
+
+		if value := findInJSON(jsonMap, h.metric.JSONField); value != nil {
```

Comment:
minor: I would say that it is more common to return `ok` as a second param to signal that the value was found or not instead of checking that the value is nil

Reply:
good point, done